README.md

```diff
@@ -0,0 +1,99 @@
+- I press the download as file button and get it returned to me as a downloadable file.
+- I exit out the page and now have my budget file
+
+User 2 (Filtered budgets view): I want to view and download certain budgets that I have created.
+
+- I open my browser and head to the budget website.
+- I hit login and enter my info.
+- I am routed to the account dashboard and select the view all budgets button.
+- I then can filter the budgets (possibly based on date created or urgency)
+- I logout of the account and exit the browser.
+
+These user stories along with mocking my webpage layout helped me very clearly define what it was I wanted to accomplish and made it easier to take small steps to get this functionality working.
+
+### Execution
+
+After the planning phase was over, I broke up the work for version one into tasks using Trello and github projects
+in order to make clear targets and goals for the foreseeable future. I used some basic concepts from agile development such as:
+
+- Having a working application with added functionality at each step of the build process
+- Having an async standup with my mentor every morning talking about what I had done yesterday and was going to do today.
+- Having pre-defined tasks that I get done over the course of a 4 - 7 day period.
+- Responding to change throughout the process such as switching to a new PDF generation package for versioning purposes.
+
+These concepts allowed me to create a workflow which would both help me effectively implement the tasks I set out to do as well as help me begin to learn about a few of these core tenets of agile development.
+Throughout this process, I gained familiarity with Flask and Python. Using Flask, I created API endpoints that would serve to either display my HTML pages or generate/save budgets.
+
+### Versioning
+
+For the versioning of the application I split up the plans I had for it into some steps for a road map. Here is a piece of the road map up until version 2:
+
+- First iteration v1.0
+  -- Quick budget generator w/file download option
+  -- Automated tests for our code
+  -- Tests can be run locally (pytest or similar)
+  -- Repo-level: use github actions to run automated tests on our branches
+  -- Create some github actions
+  -- Deployment solution: application will be containerized
+  -- Docker containers - Start off with docker-compose: we can launch our app on an env with only docker and docker compose installed, using our docker-compose script
+
+- Second iteration v2.0
+  -- Multi-user app
+  -- Ability to save and recall previously created budgets
+  -- Considerations:
+  1.  What kind of data? What is its structure? What type of DB shall we use?
+  2.  How do we do auth(orization) and (entication)?
+
+This project is still in development and there are future plans to turn it into a single page application using React for the front-end and having my Flask application serving CRUD endpoints.

```

Comment:
@Edison-Stuart I think we might want to say REST endpoints, not CRUD endpoints.

Reply:
Switched wording to REST instead of CRUD